templater/main.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/aws/eks-distro-prow-jobs/templater/jobs"
+)
+
+var jobTypes = []string{"periodic", "postsubmit", "presubmit"}
+
+var releaseBranches = []string{
+	"1-18",
+	"1-19",
+	"1-20",
+	"1-21",
+}
+
+//go:embed templates/presubmits.yaml
+var presubmitTemplate string
+
+//go:embed templates/postsubmits.yaml
+var postsubmitTemplate string
+
+//go:embed templates/periodics.yaml
+var periodicTemplate string
+
+//go:generate cp ../BUILDER_BASE_TAG_FILE ./BUILDER_BASE_TAG_FILE
+//go:embed BUILDER_BASE_TAG_FILE
+var builderBaseTag string
+
+func main() {
+	for _, jobType := range jobTypes {
+		jobList, err := jobs.GetJobList(jobType)
+		if err != nil {
+			fmt.Printf("Error getting job list: %v\n", err)
+			os.Exit(1)
+		}
+		template, err := useTemplate(jobType)
+		if err != nil {
+			fmt.Printf("Error getting job list: %v\n", err)
+			os.Exit(1)
+		}
+
+		for repoName, jobConfigs := range jobList {
+			for fileName, jobConfig := range jobConfigs {
+				data := map[string]interface{}{
+					"repoName":           repoName,
+					"prowjobName":        jobConfig.JobName,
+					"runIfChanged":       jobConfig.RunIfChanged,
+					"branches":           jobConfig.Branches,
+					"cronExpression":     jobConfig.CronExpression,
+					"maxConcurrency":     jobConfig.MaxConcurrency,
+					"timeout":            jobConfig.Timeout,
+					"extraRefs":          jobConfig.ExtraRefs,
+					"imageBuild":         jobConfig.ImageBuild,
+					"runtimeImage":       jobConfig.RuntimeImage,
+					"localRegistry":      jobConfig.LocalRegistry,
+					"serviceAccountName": jobConfig.ServiceAccountName,
+					"command":            strings.Join(jobConfig.Commands, "\n&&\n"),
+					"builderBaseTag":     builderBaseTag,
+					"resources":          jobConfig.Resources,
+					"buildkitResources":  jobConfig.BuildkitResources,
+					"envVars":            jobConfig.EnvVars,
+					"volumes":            jobConfig.Volumes,
+					"volumeMounts":       jobConfig.VolumeMounts,
+				}
+				if jobConfig.HasReleaseBranches {
+					for _, branch := range releaseBranches {
+						prowjobName := strings.ReplaceAll(jobConfig.JobName, "1-X", branch)
+						prowjobFileName := strings.ReplaceAll(fileName, "1-X", branch)
```

Comment:
This is fine with me, but I wonder if there is a more go-y template way of doing this instead?

Reply:
Done